phpBB/phpbb/members/controller/profile.php

```diff
@@ -0,0 +1,453 @@
+ *
+ */
+
+namespace phpbb\members\controller;
+
+use \phpbb\auth\auth;
+use \phpbb\config\config;
+use \phpbb\db\driver\driver_interface;
+use \phpbb\event\dispatcher;
+use \phpbb\group\helper as group_helper;
+use \phpbb\controller\helper;
+use \phpbb\language\language;
+use \phpbb\template\template;
+use \phpbb\user;
+
+class profile
+{
+	/**
+	 * @var auth
+	 */
+	protected $auth;
+
+	/**
+	 * @var config
+	 */
+	protected $config;
+
+	/**
+	 * @var driver_interface
+	 */
+	protected $db;
+
+	/**
+	 * @var dispatcher
+	 */
+	protected $dispatcher;
+
+	/**
+	 * @var group_helper
+	 */
+	protected $group_helper;
+
+	/**
+	 * @var helper
+	 */
+	protected $helper;
+
+	/**
+	 * @var language
+	 */
+	protected $language;
+
+	/**
+	 * @var template
+	 */
+	protected $template;
+
+	/**
+	 * @var user
+	 */
+	protected $user;
+
+	public function __construct(auth $auth, config $config, driver_interface $db, dispatcher $dispatcher, group_helper $group_helper, helper $helper, language $language, template $template, user $user)
+	{
+		$this->auth				= $auth;
+		$this->config			= $config;
+		$this->db				= $db;
+		$this->dispatcher		= $dispatcher;
+		$this->group_helper		= $group_helper;
+		$this->helper			= $helper;
+		$this->language			= $language;
+		$this->template			= $template;
+		$this->user				= $user;
+	}
+
+	/**
+	 * Controller for /profile/{username} route
+	 *
+	 * @return \Symfony\Component\HttpFoundation\Response a Symfony response object
+	 */
+	/*
+	public function handle_username($username)
+	{
+		// Get user...
+		$sql = 'SELECT *
+			FROM ' . USERS_TABLE . "
+			WHERE username_clean = '" . $this->db->sql_escape(utf8_clean_string($username)) . "'";
+		$result = $this->db->sql_query($sql);
+		$member = $this->db->sql_fetchrow($result);
+		$this->db->sql_freeresult($result);
+
+		return $this->handle($member['user_id']);
+	}
+	*/
+
+	/**
+	 * Controller for /profile/{user_id} route
+	 *
+	 * @return \Symfony\Component\HttpFoundation\Response a Symfony response object
+	 */
+	public function handle($user_id)
+	{
+		global $phpbb_root_path, $phpEx, $phpbb_container, $phpbb_admin_path;
+
+		// Display a listing of board admins, moderators
+		if (!function_exists('display_user_activity'))
+		{
+			include($phpbb_root_path . 'includes/functions_display.' . $phpEx);
+		}
+
+		// Load language strings
+		$this->language->add_lang('memberlist');
+
+		// Can this user view profiles/memberlist?
+		if (!$this->auth->acl_gets('u_viewprofile', 'a_user', 'a_useradd', 'a_userdel'))
+		{
+			if ($this->user->data['user_id'] != ANONYMOUS)
+			{
+				send_status_line(403, 'Forbidden');
+				trigger_error('NO_VIEW_USERS');
+			}
+
+			login_box('', $this->language->lang('LOGIN_EXPLAIN_VIEWPROFILE'));
+		}
+
+		$user_id = (int) $user_id;
+
+		// Display a profile
+		if ($user_id == ANONYMOUS)
+		{
+			trigger_error('NO_USER');
+		}
+
+		// Get user...
+		$sql = 'SELECT *
+			FROM ' . USERS_TABLE . "
+			WHERE user_id = $user_id";
```

Comment:
The former where clause looks like https://github.com/phpbb/phpbb/blob/3.2.x/phpBB/memberlist.php#L494

`WHERE ' . (($username) ? "username_clean = '" . $db->sql_escape(utf8_clean_string($username)) . "'" : "user_id = $user_id");`

since we are able to query a profile using also the username
IE.: https://www.phpbb.com/community/memberlist.php?mode=viewprofile&un=3Di

The request variable UN is located here: 
https://github.com/phpbb/phpbb/blob/3.2.x/phpBB/memberlist.php#L42
and used only by the viewprofile mode.

You are using only the `user_id` this time. Has been on purpose?

Reply:
Yes, I removed that because I think is not used, but i left that feature commented (just in case) to query by username like this:
https://www.phpbb.com/profile/u/3Di

This is the related source code:
https://github.com/phpbb/phpbb/pull/5396/files#diff-5901732cfccfb7e5dd22f5b2f25651dbR8
https://github.com/phpbb/phpbb/pull/5396/files#diff-852aca5d30ed299b71595143c4108cb8R71
https://github.com/phpbb/phpbb/pull/5396/files#diff-82af2ee7a649c1646b3128ea08246c02R92

I want to remove that since it isn't used anywhere, that feature should be implemented by any extension that implement the SEO urls

Reply:
BUT, I wish it was there in core phpBB ....

Reply:
So we won't be able to use `https://phpbb.com/profile/u/{UserName}`,
we need to implement Extension to do that ?